### PR TITLE
Fix installation and test

### LINKED ISSUE #1: Problem installing software in the repo
Thanks @Giulero for sharing this repo! I tried to run the software, but I experienced some problems,  listed in the following.

# Blocking Problems

## Impossible to clone repos

If I try to clone the repos using the command specify in the README, I get this error:
~~~
traversaro@IITICUBLAP257:~/tmp$ git clone git@github.com:ami-iit/ironcub_software.git
Cloning into 'ironcub_software'...
The authenticity of host 'github.com (140.82.121.3)' can't be established.
ED25519 key fingerprint is SHA256:+DiY3wvvV6TuJJhbpZisF/zLDA0zPMSvHdkr4UvCOqU.
This key is not known by any other names
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Warning: Permanently added 'github.com' (ED25519) to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
~~~

### Possible solution

Perhaps we can change the commands to `git clone https://github.com/ami-iit/ironcub_software`?

## Error when the python package is installed?

When I install the python package, I have this error:
~~~
(ironcubplanning) traversaro@IITICUBLAP257:~/paper_lerario_2022_humanoids_planning-multimodal-locomotion$ pip install .
Processing /home/traversaro/paper_lerario_2022_humanoids_planning-multimodal-locomotion
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: wheel in /home/traversaro/mambaforge/envs/ironcubplanning/lib/python3.11/site-packages (from multi-loco-planner==0.1.dev32) (0.38.4)
ERROR: Could not find a version that satisfies the requirement casadi (from multi-loco-planner) (from versions: none)
ERROR: No matching distribution found for casadi
~~~

### Possible solution

Install `casadi`  with mamba as well?

## Error due to wrong location of where the scripts are launched

If I go to the `dataset` directory and launch a script:
~~~
(ironcubplanning) traversaro@IITICUBLAP257:~/paper_lerario_2022_humanoids_planning-multimodal-locomotion/datasets$ python plot_data_take_off.py
Traceback (most recent call last):
  File "/home/traversaro/paper_lerario_2022_humanoids_planning-multimodal-locomotion/datasets/plot_data_take_off.py", line 11, in <module>
    with open("datasets/pickles/03-15-09-54.pickle", "rb") as f:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'datasets/pickles/03-15-09-54.pickle'
~~~

Everything works fine if the script is running as:
~~~
(ironcubplanning) traversaro@IITICUBLAP257:~/paper_lerario_2022_humanoids_planning-multimodal-locomotion$ python ./datasets/plot_data_take_off.py
~~~

### Possible solution

Document how to run the scripts in the README?

## Missing meshes

When I run the scripts, the visualization works but no meshes are shown, and the terminal is full or errors such as
~~~
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_root_link_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_root_link_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_l_hip_1_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_l_hip_1_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_lap_belt_1_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_lap_belt_1_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_hip_1_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_hip_1_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_hip_2_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_hip_2_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_upper_thigh_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_upper_thigh_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_thigh_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_thigh_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_shank_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_shank_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_ankle_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_ankle_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_foot_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_foot_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_sole_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_sole_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_lap_belt_2_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_lap_belt_2_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_chest_bb_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_chest_bb_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_neck_1_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_neck_1_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_l_p220_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_l_p220_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_chest_aruco_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_chest_aruco_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_p220_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_r_p220_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_neck_2_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_neck_2_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_head_prt.obj
Error in loading mesh package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_head_prt.obj
Could not load mesh, because file could not be opened: : package://iRonCub-Mk1_1/iRonCub/meshes/obj/sim_sea_2-5_l_shoulder_1_prt.obj
~~~

planner

### Possible solution

Which version of iDynTree are you using? We should try to understand why it is working for you.

# Nice to have

## Reduce the number of conda packages installed?

The instructions install a lot of packages, `blockfactory`, `gazebo-yarp-plugins`. However, it seems that the only one actually required (if I am not wrong) are: `icub-models yarp idyntree` ?

## Simplify the creation of the mamba environment?

You can use a single command to create the environment, this simplify the process of copy&paste, for example
~~~
mamba create -n ironcubplanning icub-models yarp idyntree casadi
~~~

## Make the installation reproducible?

At the moment the instructions provided do not pin any dependencies, meaning that they risk of stop working in the future as new version of the installed software is released. How to do that? In your case, I guess it would be a good start to:
- pin the version of the mamba packages installed
- pin the version of the packages installed via pip
- specify the tag or commit of ironcub_software used

### PR DESCRIPTION
**Pull request by Giulero**
This PR addresses the issues listed in #1.

I managed the dependencies and their version using an `environment.yml` file.
@traversaro is it a good solution? 

I also simplified the procedure of launching the data script files.
I created an application using [`Typer`](https://github.com/tiangolo/typer). Now I can run the scripts typing from the root:
```bash
show_trajectory take-off
```

Fixes #1.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+dependencies:
+  - casadi=3.5.5
+  - icub-models=1.26.0
+  - idyntree=8.1.0
+  - pip
+  - yarp=3.7.2
+  - pip:
+      - adam-robotics==0.0.6
+      - liecasadi==0.0.4
+prefix: /home/glerario/mambaforge/envs/ironcubplanning
```
Comment (traversaro) [inline environment.yml]:
```suggestion
```
Probably the prefix part can be removed?

**Comment by traversaro** [comment]
> @traversaro is it a good solution?

Sure!

File: README.md [hunk 2]
```diff
@@ -143,19 +134,44 @@ export GAZEBO_MODEL_PATH=${GAZEBO_MODEL_PATH}:${IRONCUB_SOFTWARE_SOURCE_DIR}/mod
...
+
+```bash
 cd paper_lerario_2022_humanoids_planning-multimodal-locomotion
+conda config --add channels conda-forge # if you don't have the conda-forge channel already
+mamba env create -f environment.yml
 pip install .
 ```
 
-To run the playback of the data, run the scripts in the [datasets folder](https://github.com/ami-iit/paper_lerario_2022_ral_planning-multimodal-locomotion/tree/main/datasets). It will show the visualizations of the trajectories and plots.
+Run from the root
```
Comment (traversaro) [inline README.md]:
```suggestion
Run from the root directory of the repository
```

**Review by traversaro — COMMENTED** [review]
Thanks @Giulero !

**Comment by Giulero** [comment]
Thanks @traversaro!

**Review by traversaro — APPROVED** [review]

**Comment by Giulero** [comment]
Thanks @traversaro! :)

**Pull request merged**